### ISSUE TITLE
[Bug]: build is failing for new Otter library

### ISSUE THREAD

**Issue by vscaiceanu-1a** [AmadeusITGroup/otter#1717]
### Package name

core, workspace

### Package version

10.2.1

### Reproduction steps

```shell
npm create @o3r otter-project
cd otter-project
ng g library somelib
npm run ng build somelib
```

### Current result

```shell

------------------------------------------------------------------------------
Building entry point 'somelib'
------------------------------------------------------------------------------
✖ Compiling with Angular sources in Ivy partial compilation mode.
error TS5090: Non-relative paths are not allowed when 'baseUrl' is not set. Did you forget a leading './'?
```

Adding `baseUrl` by hand leads to another error:

`Dependency @o3r/core must be explicitly allowed using the "allowedNonPeerDependencies" option.`

### Expected result

Build success

### Additional comments

The second error comes from the fact that `@o3r/core` is added as `dependency` and not `devDependency` on the generated library.

**Comment by vscaiceanu-1a** [comment]
`baseUrl` issue is fixed in #1698

**Comment by vscaiceanu-1a** [comment]
fixed in #1718
